src/functions.rs

```diff
@@ -158,205 +177,298 @@ pub mod stdlib {
-    // function.
-    pub struct StartColumn;
-
-    impl Function for StartColumn {
-        fn call(
-            &mut self,
-            graph: &mut Graph,
-            _source: &str,
-            parameters: &mut dyn Parameters,
-        ) -> Result<Value, ExecutionError> {
-            let node = parameters.param()?.into_syntax_node(graph)?;
-            parameters.finish()?;
-            Ok(Value::Integer(node.start_position().column as u32))
+    pub mod graph {
+        use super::*;
+
+        /// The implementation of the standard [`node`][`crate::reference::functions#node`] function.
+        pub struct Node;
+
+        impl Function for Node {
+            fn call(
+                &mut self,
+                graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                parameters.finish()?;
+                let node = graph.add_graph_node();
+                Ok(Value::GraphNode(node))
+            }
         }
     }
 
-    // The implementation of the standard [`end-row`][`crate::reference::functions#end-row`]
-    // function.
-    pub struct EndRow;
-
-    impl Function for EndRow {
-        fn call(
-            &mut self,
-            graph: &mut Graph,
-            _source: &str,
-            parameters: &mut dyn Parameters,
-        ) -> Result<Value, ExecutionError> {
-            let node = parameters.param()?.into_syntax_node(graph)?;
-            parameters.finish()?;
-            Ok(Value::Integer(node.end_position().row as u32))
+    pub mod bool {
+        use super::*;
+
+        /// The implementation of the standard [`not`][`crate::reference::functions#not`] function.
+        pub struct Not;
+
+        impl Function for Not {
+            fn call(
+                &mut self,
+                graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                let result = !parameters.param()?.into_bool(graph)?;
+                parameters.finish()?;
+                Ok(result.into())
+            }
         }
-    }
 
-    // The implementation of the standard [`end-column`][`crate::reference::functions#end-column`]
-    // function.
-    pub struct EndColumn;
+        /// The implementation of the standard [`and`][`crate::reference::functions#and`] function.
+        pub struct And;
+
+        impl Function for And {
+            fn call(
+                &mut self,
+                graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                let mut result = true;
+                while let Ok(parameter) = parameters.param() {
+                    result &= parameter.into_bool(graph)?;
+                }
+                Ok(result.into())
+            }
+        }
```

Comment:
I have the strangest feeling of déjà vu... :thinking:

Reply:
Not sure what you're hinting at here...

Reply:
Nothing important. Just that these are the exact same functions I implemented in #15. :laughing: